### PR TITLE
Add list-bin support for scripts.

### PR DESCRIPTION
**Pull request by bacchanalia**
Small PR to add script support for list-bin. It includes a test.
Because of #7774, I do not  believe there is any documentation that needs to be updated.

Related: #7073

### CODE REVIEW

**Review by Mikolaj — APPROVED** [review]
Splendid.

**Review by jneira — APPROVED** [review]
perfect companion of your main pr improving cabal script, including a regression test, thanks!

**Comment by andreasabel** [comment]
@bacchanalia wrote:
> Small PR to add script support for list-bin. It includes a test. 

I checked the [test output](https://github.com/haskell/cabal/pull/7925/files#diff-3d6637ed4ab2ee658f06eaa968642fc10f891f059a167e3a927f1908a623b1fd) but it does not contain the path to the binary.  I'd expect that `list-bin` produces this path.  Is everything working correctly?

> Because of #7774, I do not believe there is any documentation that needs to be updated.

There is [documentation of `list-bin`](https://cabal.readthedocs.io/en/stable/cabal-commands.html#cabal-list-bin) now, but does not include your use case with scripts.  This is
- #7986

Would you be able to supply this documentation now to close #7986?

**Comment by Mikolaj** [comment]
Indeed I missed the test output problem and the above suggestions make sense to me.

File: cabal-testsuite/PackageTests/ListBin/Script/cabal.test.hs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Test.Cabal.Prelude
+
+main = cabalTest . void $ do
+    res <- cabal' "list-bin" ["script.hs"]
+
+    env      <- getTestEnv
+    cacheDir <- getScriptCacheDirectory $ testCurrentDir env </> "script.hs"
+    assertOutputContains cacheDir res
```
Comment (andreasabel) [inline cabal-testsuite/PackageTests/ListBin/Script/cabal.test.hs:8]: @bacchanalia : I don't understand how this `assertOutputContains` succeeds with above output (that does not contain something resembling a `cacheDir`).  But apparently it does.  How?

**Pull request merged**